library/alloc/src/vec/mod.rs

```diff
@@ -1248,6 +1271,16 @@ impl<T, A: Allocator> Vec<T, A> {
     /// Modifying the vector may cause its buffer to be reallocated,
     /// which would also make any pointers to it invalid.
     ///
+    /// This method guarantees that when it is called multiple times without
+    /// the buffer being reallocated in the mean time, the returned pointer will
+    /// always be exactly the same, even for the purpose of the aliasing model, where
```

Comment:
As I said later, I don't think we want to commit to "exactly the same". This can all be expressed in terms of which other operations invalidate the pointer, I thought that was the plan?

Reply:
I tried to fix the language. I'm not really sure how expansive to be here.